From: Antonio Alvarez Feijoo <antonio.feijoo@suse.com>
Date: Thu, 6 Aug 2026 12:41:48 +0200
Subject: fix(net-lib): validate iSCSI port parameters

Similar to 05c02db994ca322371a40695a3cc223faa1c7258, the iSCSI port is a number.
Validate that given ports are numbers.

Origin: upstream, https://github.com/dracut-ng/dracut/commit/c8bf7dc5428f0db589e7a73a0114fbb89cd957fa
---
 modules.d/45net-lib/net-lib.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/45net-lib/net-lib.sh b/modules.d/45net-lib/net-lib.sh
index 1c9ff7d..d67cdfc 100755
--- a/modules.d/45net-lib/net-lib.sh
+++ b/modules.d/45net-lib/net-lib.sh
@@ -446,7 +446,7 @@ parse_iscsi_root() {
     [ $# -ge 2 ] || return 0
     iscsi_protocol=$1
     shift # ignored
-    iscsi_target_port=$1
+    iscsi_target_port=$(strip_non_digits "$1" "iSCSI port")
     shift
 
     if [ -n "$iscsi_target_name" ]; then
